lib/buffer.js

```diff
@@ -58,7 +58,7 @@ createPool();
 
 
 function SlowBuffer(length) {
-  if (length < 0)
+  if (length < 0 || typeof length !== 'number')
```

Comment:
Shouldn't type be checked first?  Would short circuit quicker for invalid types.

      if (typeof length !== 'number' || length < 0)